packages/evm/src/evm.ts

```diff
@@ -268,7 +328,7 @@ export class EVM implements EVMInterface {
       return {
         execResult: {
           gasRefund: message.gasRefund,
-          executionGasUsed: BIGINT_0,
+          executionGasUsed: message.gasLimit - gasLimit,
```

Comment:
Just reviewing the PR a bit in the light of non-verkle (side) changes to see what the overall effects on our libraries are/would be.

What is this change? (and also some additional gas related changes below)

Is this a general bug fix for the EVM? To now pass some dynamic value instead of `BIGINT_0`?

Reply:
this is because even in this early return case, access gas could be consumed (which is dedcted from this function scoped `gasLimit`, so the diff is the whats used)